hw/ip/otbn/doc/_index.md

```diff
@@ -361,6 +361,66 @@ Both use the same state for tracking control flow.
 This is a stack of tuples containing a loop count, start address and end address.
 The stack has a maximum depth of eight and the top of the stack is the current loop.
 
+### Loop nesting
+
+OTBN permits nested loops and branch and jumps inside loops.
+The programmer must ensure loops are *well nested*.
+For a loop to be *well nested* the following conditions must be met:
+
+* Control flow of a loop must eventually return to its end instruction without passing the end instruction of any outer loop.
+* Nested loops cannot end on the same instruction.
+
+The following loops are *well nested*:
+
+```
+LOOP x2, 3
+  LOOP x3, 1
+    ADDI x4, x4, 1
+  # NOP required so outer and inner loop end on different instructions
```

Comment:
Maybe `# The NOP ensures that ...`? Otherwise it sounds a bit like "requiring the NOP" does something.